src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -187,6 +186,16 @@ public static object CreateEnumerableValue(ref ReadStack state)
         {
             JsonPropertyInfo jsonPropertyInfo = state.Current.JsonPropertyInfo;
 
+            if (jsonPropertyInfo == null)
```

Comment:
I've moved this check to `HandleStartArray` since the context is clearer there. I no longer perform the check for dictionary because we wouldn't have gotten to `HandleStartDictionary` if we were not processing a dictionary. (The invocation of `HandleStartArray` and `HandleStartDictionary` from the main loop follow different patterns which I don't want to modify in this PR).

Ultimately, we perform this check once, so we don't need a helper anymore.